panther_analysis_tool/schemas.py

```diff
@@ -134,6 +134,7 @@
                 Optional("Mocks"): object,
             }
         ],
+        Optional("LastUpdatedVersion"): str,
```

Comment:
Do we need this here? Is the workflow for a user to pass in the cli option (vs updating their metadata files)? or are we supporting both?

further, would we want to allow specifying this in yml files? or is this something we should control based on the panther version on the backend? what if a user specifies a version and when uploading it doesn't match the panther version? do we take the one specified or force it to the panther version?

Reply:
> Do we need this here?

Yes, otherwise our unit tests will fail due to schema validation.

> Is the workflow for a user to pass in the cli option (vs updating their metadata files)? or are we supporting both?

We are supporting both -- from v1.0.0 and after, users will have to have their detection metadata files updated for the new mocking or have the `--use-legacy-mocks` flag specified. However that being said, once users have their detections updated to support the new mocking, they will have to have the `LastUpdatedVersion` added to the detection metadata. We should include these instructions in the migration notes.


> or is this something we should control based on the panther version on the backend?  what if a user specifies a version and when uploading it doesn't match the panther version? do we take the one specified or force it to the panther version?

This will be controlled on the BE. In the case where a user specifies a version that doesn't match the panther version, we'll just overwrite that version unless the version is less than the minimum version required to support the new unit test mocking.

Reply:
So for existing detections, if `LastUpdatedVersion` is `nil` then we won't update the version and it'll be processed as a legacy mock. This comes with the caveat that users attempting to update detections will be required to update their unit test mocks to adhere to the new unit test mocking. 

To alleviate the burden of having customers update all of their detections after upgrading to v1.26.0, we could include a dummy version when the `--use-legacy-mocks` flag is `True` as to avoid breaking existing CI/CD pipelines. I'll push up changes to include that feature tomorrow

Reply:
Just wondering, why couldn't we provide the hard-coded `LEGACY_MOCKING_VERSION` if the user chooses to use JSON mocks? Otherwise, just pass something like '0.0.0` to fall back?

The `LastUpdatedVersion` field in YAML detection metadata seems as not something that users can easily maintain, it is information set by the backend. In fact, it could potentially create some confusion during the upload, as you would have this field as part of input.

Reply:
> The LastUpdatedVersion field in YAML detection metadata seems as not something that users can easily maintain, it is information set by the backend. In fact, it could potentially create some confusion during the upload, as you would have this field as part of input.

@georgepsarakis That's a good point! So would we have something like a toggle in the UI that allows a user select whether they want to use the JSON mocks?